tests/rest/admin/test_user.py

```diff
@@ -1092,8 +1092,10 @@ def prepare(self, reactor: MemoryReactor, clock: Clock, hs: HomeServer) -> None:
 
         self.other_user = self.register_user("user", "pass", displayname="User1")
         self.other_user_token = self.login("user", "pass")
-        self.url_other_user = "/_synapse/admin/v2/users/%s" % urllib.parse.quote(
-            self.other_user
+        self.url_users = "/_synapse/admin/v2/users"
+        self.url_other_user = "%s/%s" % (
+            self.url_users,
+            urllib.parse.quote(self.other_user),
```

Comment:
Agreed with Dirk: we're testing the behaviour of List Users endpoint here, not the deactivate account test case. Given that, it's okay to use Synapse's internals to deactivate the account here if you prefer, see e.g. https://github.com/matrix-org/synapse/blob/4283bd1cf9c3da2157c3642a7c4f105e9fac2636/tests/rest/client/test_rooms.py#L3013-L3020